extension/chrome/extension.js

```diff
@@ -53,70 +53,76 @@ const addBookmark = async (url, tabId) => {
 }
 
 
-// Add library icons for added games
+// Add icons for games and bookmarks
 const updateIcons = async (tabId) => {
     await getData();
     chrome.scripting.executeScript({
         target: { tabId: tabId },
-        func: (games) => {
+        func: (games, bookmarks) => {
             const threadId = (url) => {
                 const match = /threads\/(?:(?:[^\.\/]*)\.)?(\d+)/.exec(url);
                 return match ? parseInt(match[1]) : null;
             }
-            const createIcon = (isImage) => {
+            const createContainer = () => {
+                const c = document.createElement("div");
+                c.classList.add("f95checker-library-icons");
+                c.style.display = "inline-flex";
+                c.style.gap = "5px";
+                c.style.padding = "3px 3px";
+                return c;
+            }
+            const gamesIcon = () => {
                 const icon = document.createElement("i");
-                icon.classList.add("fa", "fa-box-heart", "f95checker-library-icon");
                 icon.style.fontFamily = "'Font Awesome 5 Pro'";
-                icon.style.color = "#FD5555";
-                if (isImage) {
-                    icon.style.position = "absolute";
-                    icon.style.zIndex = "50";
-                    icon.style.right = "5px";
-                    icon.style.top = "5px";
-                    icon.style.fontSize = "larger";
-                    icon.style.background = "#262626";
-                    icon.style.border = "solid #262626";
-                    icon.style.borderWidth = "4px 5px";
-                    icon.style.borderRadius = "4px";
-                } else {
-                    icon.style.marginRight = "0.2em"
-                }
+                icon.classList.add("fa", "fa-box-heart");
                 icon.setAttribute("title", "This game is present in your F95Checker library!");
-                icon.addEventListener("click", () => {
-                    alert("This game is present in your F95Checker library!");
-                });
+                icon.addEventListener("click", () => alert("This game is present in your F95Checker library!"));
+                icon.style.color = "#FD5555";
+                return icon;
+            }
+            const bookmarksIcon = () => {
+                const icon = document.createElement("i");
+                icon.style.fontFamily = "'Font Awesome 5 Pro'";
+                icon.classList.add("fa", "fa-sticky-note");
+                icon.setAttribute("title", "Thread bookmarked");
+                icon.addEventListener("click", () => alert("Thread bookmarked"));
+                icon.style.color = "#55eecc";
                 return icon;
             }
             const doUpdate = () => {
-                for (elem of document.getElementsByClassName("f95checker-library-icon")) {
```

Comment:
This is the line that caused icon duplication, for some reason `for` loop does not remove all elements, but `forEach` does.

Reply:
oh, interesting